dissect/target/plugins/apps/webservers/nginx.py

```diff
@@ -0,0 +1,72 @@
+import re
+from datetime import datetime
+from pathlib import Path
+from typing import Iterator
+
+from dissect.target import plugin
+from dissect.target.helpers.fsutil import open_decompress
+from dissect.target.plugins.apps.webservers.webservers import WebserverRecord
+from dissect.target.target import Target
+
+LOG_REGEX = re.compile(
+    r'(?P<remote_ip>.*?) - (?P<remote_user>.*?) \[(?P<datetime>\d{2}\/[A-Za-z]{3}\/\d{4}:\d{2}:\d{2}:\d{2} (\+|\-)\d{4})\] "(?P<url>.*?)" (?P<status_code>\d{3}) (?P<bytes_sent>\d+) (["](?P<referer>(\-)|(.+))["]) "(?P<useragent>.*?)"',  # noqa: E501
+    re.IGNORECASE,
+)
+
+
+class NginxPlugin(plugin.Plugin):
+    __namespace__ = "nginx"
+
+    def __init__(self, target: Target):
+        super().__init__(target)
+        self.log_paths = self.get_log_paths()
+
+    @plugin.internal
+    def get_log_paths(self) -> list[Path]:
+        log_paths = []
+        default_logs_dir = self.target.fs.path("/var/log/nginx")
+        if default_logs_dir.exists():
+            log_paths = [self.target.fs.path(p) for p in default_logs_dir.glob("access.log*")]
+
+        # Check for custom paths in nginx install config
+        if (config_file := self.target.fs.path("/etc/nginx/nginx.conf")).exists():
```

Comment:
Can you add a unit test for parsing nginx log files?

Reply:
Added in [58c49a2](https://github.com/fox-it/dissect.target/pull/127/commits/58c49a2a053d4fe28b4d10ed4bd3b449c0e98219).